cmd/tapsim/main.go

```diff
@@ -317,3 +369,81 @@ func execute(cCtx *cli.Context) error {
 	fmt.Printf("script execution verified\r\n")
 	return nil
 }
+
+func validate(cCtx *cli.Context) error {
```

Comment:
Similar to the other comment, I think this shouldn't be its own command, but be merged with `execute`.

Maybe by adding `--tx` and `--prevouts` arguments to the existing command. 

That way one could use the `--tx` argument to specify a template for a tx (the template could of course also be a fully signed and valid tx), while the other arguments can be used to override certain parts of the template (script, witness, inputs, outputs etc).